README.md

```diff
@@ -1,6 +1,218 @@
 # CNS Object Broker
-## For dynmically provisioning object storage compliant with an S3 interface.
+## Utilize Kubernetes Service-Catalog to dynamically provision CNS Object Storage.
 
 ## WARNING!
+The work in this project is a proof of concept and not intended production.
 
-The work in this project is a proof of concept and not intended for anything more than that.
+## Overview
+A core feature of the Kubernetes system is the ability to provision a diverse
+offering of block and file storage on demand.  This project seeks to demonstrate
+that by using the Kubernetes-Incubator's Service-Catalog, it is now also possible
+to bring this dynamic provisioning to S3 object storage.  This broker is designed
+to be used with [Gluster-Kubernetes](https://github.com/jarrpa/gluster-kubernetes).
+
+See our detailed [command flow diagram](./docs/Service-Catalog-Published.png) for
+ an overview of the system.
+
+### What It Does
+This broker implements OpenServiceBroker methods for creating, connecting to, and destroying
+object buckets.
+- For each new Service Instance, a new, unquiely named bucket is created.
+- For each new Service Instance Credential, a Secret is generated with the
+coordinates and credentials of the Service Instance's associated bucket.
+- Deleting a Service Instance destroys the associated bucket.
+- Deleting a Service Instance Credential deletes the secret.  Nothing is done on the broker side.  This is because the broker does not perform any actions on Bind, so there is nothing to
+undo.
+
+## Installation
+
+### Dependencies
+- [Google SDK](https://cloud.google.com/sdk/) (`gcloud`): To access GCP via cli.
+- [Kubernetes](https://github.com/kubernetes/kubernetes): to run local K8s cluster
+- [Gluster-Kubernetes Deploy Script](https://github.com/copejon/gk-cluster-deploy): to deploy GCP K8s cluster with Gluster-Kubernetes Object Storage and CNS Object Broker
+- Kubernetes-Incubator [Service-Catalog](https://github.com/kubernetes-incubator/service-catalog): to provide Service-Catalog helm chart
+- Kubernetes [Helm](https://github.com/kubernetes/helm#install): to install Service-Catalog chart
+
+### Assumptions
+- Gk-cluster-deploy was written and tested on Fedora 25 and RHEL 7.
+- A [Google Cloud Platform](https://cloud.google.com/compute/) account.
+
+### Topology
+There are two primary systems that make up this demonstration.  They are the Broker
+and its colocated CNS Object store.  They fill the role of our micro-service provider.
```

Comment:
"... its colocated CNS Object store ..."
Consider adding a sentence that mentions that, in general, the s3-brokers and object store do _not_ have to reside on a k8s cluster but they are in this example. And also this cluster can be on-prem.